pkg/extensions/extension_api_key.go

```diff
@@ -0,0 +1,209 @@
+//go:build apikey
+// +build apikey
+
+package extensions
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	guuid "github.com/gofrs/uuid"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/sessions"
+	jsoniter "github.com/json-iterator/go"
+	godigest "github.com/opencontainers/go-digest"
+
+	"zotregistry.io/zot/pkg/api/config"
+	"zotregistry.io/zot/pkg/api/constants"
+	"zotregistry.io/zot/pkg/log"
+	"zotregistry.io/zot/pkg/meta/repodb"
+	localCtx "zotregistry.io/zot/pkg/requestcontext"
+)
+
+func SetupAPIKeyRoutes(config *config.Config, router *mux.Router, repoDB repodb.RepoDB,
+	cookieStore *sessions.FilesystemStore, log log.Logger,
+) {
+	if config.Extensions.APIKey != nil && *config.Extensions.APIKey.Enable {
+		log.Info().Msg("setting up user preferences routes")
+
+		apiKeyRouter := router.PathPrefix(constants.ExtAPIKeyPrefix).Subrouter()
+
+		apiKeyRouter.HandleFunc("", HandleAPIKeyRequest(repoDB, cookieStore, log)).Methods(http.MethodPost, http.MethodDelete)
+	}
+}
+
+type APIKeyPayload struct { //nolint:revive
+	Label  string   `json:"label"`
+	Scopes []string `json:"scopes"`
+}
+
+func HandleAPIKeyRequest(repoDB repodb.RepoDB, cookieStore *sessions.FilesystemStore,
+	log log.Logger,
+) func(w http.ResponseWriter, r *http.Request) {
+	return func(resp http.ResponseWriter, req *http.Request) {
+		switch req.Method {
+		case http.MethodPost:
```

Comment:
We'll need to also implement a case for MethodOptions.

Reply:
yes